crates/view/src/storage.rs

```diff
@@ -1169,11 +1169,6 @@ impl Storage {
     }
 
     pub async fn give_advice(&self, note: Note) -> anyhow::Result<()> {
-        // Do not insert advice for zero amounts, simply return Ok because this is fine
-        if u128::from(note.amount()) == 0u128 {
-            return Ok(());
-        }
```

Comment:
wow really nice find! Two thoughts:
1. I think we'd need to apply this same change in the view impl over in `IndexedDBStorage::store_advice` in the wasm crate, such that the same fix is applied over there
2. The implication of this change is that #2136 is re-introduced, so maybe we do something like filter out 0-valued notes when querying for spendable notes (then users won't see behavior like https://discord.com/channels/824484045370818580/915710988312002610/1087796172078469281). What do you think?

Reply:
Confirmed that this reintroduces the issue mentioned previously in discord:

```Scanning blocks from last sync height 156681 to latest height 160295
[0s] ██████████████████████████████████████████████████    3614/3614    4133/s ETA: 0s
 Account  Amount      
 0        1gm         
 0        0gm         
 0        9gm         
 0        0gm         
 0        0gm         
 0        996test_usd 
 0        1test_usd   
 0        5penumbra   
 0        4penumbra   
 0        0gn         
 0        0gn         
 0        7gn         
 0        1gn         
 0        1gn         
 0        1gn         
 0        30penumbra  
aubrey@penpro penumbra % 
```

I like your suggestion - the balances query needs fixing anyway (it's showing several rows for one denom instead of properly consolidating denoms, as in #3214 ), so it seems like not including zero-valued notes as "spendable" is something we always should have been doing. Good catch on also doing this in the wasm crate - I'll dig into this tomorrow, and if we're lucky this will also shed some light on the existing balances query issue. Thank you for the review!